### PR TITLE
Added placeholder and spinner when loading

### PR DESCRIPTION
**Pull request by LukasHirt**
## Description
Added spinner when avatar is being loaded and placeholder if the src string is empty. Also added flexbox for fixing the alignment for the placeholder.

## Related issue
Fixes: #258 and https://github.com/owncloud/phoenix/issues/971

### CODE REVIEW

File: src/elements/OCAvatar.vue [hunk 1]
```diff
@@ -10,6 +22,13 @@ export default {
   status: "review",
   release: "1.0.0",
   props: {
+    /**
+     * Defines if an avatar has been loaded
+     */
+    loaded: {
+      type: Boolean,
+      required: false,
```
Comment (DeepDiver1975) [inline src/elements/OCAvatar.vue]: add default false

Comment (DeepDiver1975) [reply src/elements/OCAvatar.vue]: and loading might sounds more reasonable

File: src/elements/OCAvatar.vue [hunk 2]
```diff
@@ -36,9 +55,34 @@ export default {
...
 ```jsx
-<div>
-  <oc-avatar src="https://picsum.photos/64/64?image=1074" title="I'm a Lion"/>
-  <oc-avatar src="https://picsum.photos/64/64?image=718"/>
-</div>
+<template>
+  <div class="uk-flex uk-flex-middle">
+    <oc-avatar :loaded="loading" :src="srcLion" title="I'm a Lion"  />
+    <oc-avatar :loaded="loading" :src="srcWolf" />
+    <oc-avatar :loaded="loading" :src="srcNonEmpty" title="I'm using the placeholder" />
```
Comment (DeepDiver1975) [inline src/elements/OCAvatar.vue]: srcNonEmpty does not exist

File: src/elements/OCAvatar.vue (hunk 1 above)
Comment (LukasHirt) [reply src/elements/OCAvatar.vue]: Right - fixed.

File: src/elements/OCAvatar.vue (hunk 2 above)
Comment (LukasHirt) [reply src/elements/OCAvatar.vue]: 🤦‍♂ fixed

**Comment by LukasHirt** [comment]
> Screenshot from 2019-05-02 08-05-36
> 
> Looks like [owncloud/phoenix#971](https://github.com/owncloud/phoenix/issues/971) is not yet fully resolved....

Now it should be fixed with the width and height classes. I also added them to oc-spinner because he also looked a little bit off in the files.

**Comment by LukasHirt** [comment]
> [Vue warn]: Missing required prop: "src"
> 
> found in
> 
> ---> at src/elements/OCAvatar.vue

Fixed

**Comment by DeepDiver1975** [comment]
> Fixed

did you forget to push? I still see this in the error console ...

File: src/pages/Files.vue [hunk 3]
```diff
@@ -72,6 +72,7 @@ export default {
           <oc-avatar
             class="oc-topbar-personal-avatar"
             src="https://picsum.photos/64/64?image=1027"
```
Comment (LukasHirt) [inline src/pages/Files.vue]: It's here @DeepDiver1975 It was missing in OCAvatar at the one with placeholder but that is fixed already and pushed.

Comment (DeepDiver1975) [reply src/pages/Files.vue]: The point is: if src is not *yet* set we will get this warning in phoenix as well. So it should be marked as required: false

Comment (LukasHirt) [reply src/pages/Files.vue]: Done and the test passes so I guess it should be fine now.

**Pull request merged**